lib/fluent/plugin/formatter_out_file.rb

```diff
@@ -43,10 +43,13 @@ def configure(conf)
       end
 
       def format(tag, time, record)
+        json_str = JSON.generate(record)
         header = ''
         header << "#{@timef.format(time)}#{@delimiter}" if @output_time
         header << "#{tag}#{@delimiter}" if @output_tag
-        "#{header}#{JSON.generate(record)}#{@newline}"
+        "#{header}#{json_str}#{@newline}"
+      ensure
+        json_str.clear
```

Comment:
```suggestion
        result = "#{header}#{json_str}#{@newline}"
        json_str.clear
        result
```

I'm concerned that `json_str` can be `nil`.
There would be 2 cases:

* `JSON.generate` raises an exception.
* The process is killed during `JSON.generate`.
  * I encountered this case during testing performance.
    * ``[warn]: #0 emit transaction failed: error_class=NoMethodError error="undefined method `clear' for nil:NilClass"``

It would not be a problem if `String#clear` cannot be performed.
So, can we remove `ensure`?

Reply:
Indeed.

I will fix that using `&.`.